sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -372,12 +372,44 @@ class HiveCatalogedDDLSuite extends DDLSuite with TestHiveSingleton with BeforeA
       assert(table.provider == Some("org.apache.spark.sql.hive.orc"))
     }
   }
+
+  test("Database Ownership") {
+    val catalog = spark.sessionState.catalog
+    try {
+      val dbName = "spark_29425"
+      val location = getDBPath(dbName)
+
+      sql(s"CREATE DATABASE $dbName")
+
+      checkAnswer(
+        sql(s"DESCRIBE DATABASE $dbName"),
+        Row("Database Name", dbName) ::
+          Row("Description", "") ::
+          Row("Location", CatalogUtils.URIToString(location)) ::
+          Row("Owner Name", Utils.getCurrentUserName()) ::
+          Row("Owner Type", "USER") :: Nil)
+
+      sql(s"ALTER DATABASE $dbName SET DBPROPERTIES ('a'='a', 'b'='b', 'c'='c')")
```

Comment:
Just to confirm: when a user runs the command `ALTER DATABASE SET DBPROPERTIES`, we will reset owner to the default value if the DBPROPERTIES doesn't contain owner?

Reply:
Behavior Before: we `create db` with no owner, `alter db` erase the owner if exists not with our default spark user.
Behavior After: we `create db` with the spark user as default, or with dbProps if ownerName exists; we `alter db` will prefer the owner in order of  `specified ownerName` -> `original db's ownerName` -> `spark's default if the foregoing ones are null or empty`.

`ALTER DATABASE dbname SET DBPROPERTIES('ownerName'='userName')` equals Hive's `ALTER DATABASE dbname SET OWNER USER userName`.

I suggest we make the ownerName and its type become members of `CatalogDabase` in followup, and the `CatalogTable` too.
Then support，
```sql
ALTER [DATABASE|SCHEMA] dbname SET OWNER [USER|ROLE] userName
ALTER TABLE tblname SET OWNER [USER|ROLE] userName
```

Reply:
we should definitely add `ALTER DATABASE dbname SET OWNER USER userName`, but I'm not sure it's the right approach to keep adding fields to `CatalogDatabase`/`CatalogTable` in the long term.

`CatalogDatabase`/`CatalogTable` are private so we can change them. But in the long term, we will have a stable catalog API (e.g. `TableCatalog`), and at that time it's not allowed to add more fields in newer versions.

Can we add tests to verify the `Behavior After`?

Reply:
DCL support or ACL Management for Spark SQL is a necessary feature in the long term, I guess. And these fields have been stable in hive metastore API for years, we may consider following that in our own catalog API. 

BTW, we already have tests for `Behavior After` in `VersionsSuite`.

Reply:
I'm not talking about this specific case but the general framework. What if we need to add "lastAccessTime" later? We can't keep adding new fields and break the API. Using properties is more future-proof.

Reply:
In this case, we may reverse some fields for security,  `ALTER DATABASE dbname SET DBPROPERTIES('ownerName'='userName')` should be forbidden,  because it hive different OperationType or PrivillegeType from `SET Owner`syntax in Hive

Reply:
yea we can define some private properties which can't be changed with `SET DBPROPERTIES `